platform-sdk/swirlds-common/src/testFixtures/java/com/swirlds/common/test/fixtures/merkle/util/MerkleTestUtils.java

```
@@ -1083,6 +1084,13 @@ public static <T extends MerkleNode> T testSynchronization(
             assertReconnectValidity(startingTree, desiredTree, generatedTree);
 
             return (T) generatedTree;
+        } finally {
+            try {
+                streams.close();
+            } catch (IOException e) {
+                // test code, no danger. The stream could have been closed previously.
+                e.printStackTrace();
+            }
```

A safer approach would be to check if the stream is closed before calling `close()`. This would prevent the test from swallowing exceptions when it shouldn't swallow them.

---

Alternatively, the stream implementation shouldn't error-out when `close()` is called on an already closed stream. This would fix the root cause of the issue instead of implementing workarounds such as suggested above or presented in the current commit here.

---

Fair enough. I moved exception handling to `PairedStreams`. It's a test fixture anyway, so I think it should be fine.